File: Data Structures & Algorithms/duplicate-integer/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    //nums = [1, 2, 3, 3]
    hasDuplicate(nums) {
       const seen = new Set()
       for (let i = 0; i <nums.length; i++){
        const val = nums[i]
        if (seen.has(val)){
            return true
        }
        seen.add(val)
       }
       return false 
}
}

File: Data Structures & Algorithms/is-anagram/submission-2.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        map = {}
        for char in s: 
            map[char] = map.get(char,0) + 1 # r2 a2 c2 e1
        for char in t:
            if char not in map:
                return False
            map[char] = map.get(char,0) - 1
            if map.get(char,0) < 0:
                return False
        return True
